Libplanet.Tests/Blockchain/BlockChainTest.cs

```diff
@@ -2005,10 +2005,82 @@ private async Task FilterLowerNonceTxAfterStaging()
             Assert.Equal(4, _blockChain.StagePolicy.Iterate(_blockChain, filtered: false).Count());
         }
 
+        [Fact]
+        private void CheckIfTxPolicyExceptionHasInnerException()
+        {
+            IValue value = null;
+            var policy = new NullPolicyButTxPolicyAlwaysThrows<DumbAction>(
+                c =>
+                {
+                    // ReSharper disable AccessToModifiedClosure
+                    // The following method calls should not throw any exceptions:
+                    c?.GetStates(new[] { default(Address) });
+                    value = c?.GetState(default);
+                    // ReSharper restore AccessToModifiedClosure
+                });
+            var store = new MemoryStore();
+            var stateStore = new TrieStateStore(new MemoryKeyValueStore());
+            Block<DumbAction> genesisWithTx = MineGenesis(
+                policy.GetHashAlgorithm,
+                GenesisMiner.PublicKey,
+                new[]
+                {
+                    Transaction<DumbAction>.Create(
+                        0,
+                        new PrivateKey(),
+                        null,
+                        Array.Empty<DumbAction>()
+                    ),
+                }
+            ).Evaluate(GenesisMiner, policy.BlockAction, stateStore);
+
+            try
+            {
+                var chain = new BlockChain<DumbAction>(
+                    policy,
+                    new VolatileStagePolicy<DumbAction>(),
+                    store,
+                    stateStore,
+                    genesisWithTx
+                );
+            }
+            catch (TxPolicyViolationException e)
+            {
+                Assert.NotNull(e.InnerException);
+
+                // This is visual representation of the test.
+                // In actual use-case, this debugging lines do
+                // not show.
+                _logger.Debug(e.ToString());
+                _logger.Debug(e.InnerException.ToString());
```

Comment:
The first parameter of logging methods take a template string, rather than a bare string, which means it can misbehave if its first argument contains some placeholders like `{0}`.  I know `e.ToString()` and `e.InnerException.ToString()` probably does not contains such things at the moment, someone might change these methods in the future so that their return values contain such placeholder syntax.

```suggestion
                _logger.Debug("Exception: {0}", e.ToString());
                _logger.Debug("InnerException: {0}", e.InnerException.ToString());
```